Add topicalness facet type

MatMoore opened the pull request:
This is a new kind of facet to dynamically split topical events into open or closed based on the end date.

This will look something like this when the finder itself is published:
<img width="990" alt="screen shot 2016-11-24 at 15 05 14" src="https://cloud.githubusercontent.com/assets/87579/20628227/a71ec030-b31c-11e6-96be-72f449eedfa2.png">

I've also removed what seems like an unused field from the presenters, and fixed some broken tests. This should not affect existing finders (I hope).

Trello: https://trello.com/c/oVfSAO9W/311-migrate-policy-areas-topical-events-index-to-a-finder
Matching schema change: https://github.com/alphagov/govuk-content-schemas/pull/441
Finder: https://github.com/alphagov/whitehall/pull/2855

tijmenb commented on app/lib/filter_query_builder.rb:
```diff
@@ -23,6 +23,7 @@ def build_filter(facet)
     filter_class = {
       'date' => DateFilter,
       'text' => TextFilter,
+      'topical' => TopicalFilter,
```
https://github.com/alphagov/finder-frontend/blob/master/docs/finder-content-item.md#type needs updating for this I think.

tijmenb commented on app/lib/filter_query_builder.rb:
```diff
@@ -86,4 +87,41 @@ def value
+      elsif closed && !open
+        closed_value
+      else
+        # with both selected, the filter is not used
```
This should probably read "with both or neither selected".

MatMoore commented:
@tijmenb Have addressed your comments, except for the docs, which I'll add in a separate PR.

If this and https://github.com/alphagov/whitehall/pull/2855 are good to merge I can deploy the topical events finder this morning.
